File: src/test/java/com/dotnomi/fabricdependencyinjection/java/testclasses/circular/CircularA.java
package com.dotnomi.fabricdependencyinjection.java.testclasses.circular;

import com.dotnomi.fabricdependencyinjection.annotation.ModInject;
import com.dotnomi.fabricdependencyinjection.annotation.ModScoped;

@ModScoped
public class CircularA {
  @ModInject
  public CircularA(CircularB circularB) {}
}

File: src/test/java/com/dotnomi/fabricdependencyinjection/kotlin/testclasses/circular/CircularA.kt
package com.dotnomi.fabricdependencyinjection.kotlin.testclasses.circular

import com.dotnomi.fabricdependencyinjection.annotation.ModInject
import com.dotnomi.fabricdependencyinjection.annotation.ModScoped

@ModScoped
class CircularA @ModInject constructor(circularB: CircularB)
